src/fontra/client/core/actions.js

```diff
@@ -27,19 +28,22 @@ actionInfoController.addListener((event) => {
   actionsByKeyOrCode = undefined;
 });
 
+let sortIndex = 0;
 export function registerAction(actionIdentifier, actionInfo, callback, enabled = null) {
-  registerActionInfo(actionIdentifier, actionInfo);
+  registerActionInfo(actionIdentifier, actionInfo, sortIndex);
   registerActionCallbacks(actionIdentifier, callback, enabled);
+  sortIndex++;
 }
 
-export function registerActionInfo(actionIdentifier, actionInfo) {
+export function registerActionInfo(actionIdentifier, actionInfo, sortIndex) {
   actionInfoController.synchronizeItemWithLocalStorage(actionIdentifier, actionInfo);
   // We only want customShortCuts to be changable, so we'll reset everything
   // except customShortCuts
   const storedActionInfo = getActionInfo(actionIdentifier);
   actionInfoController.model[actionIdentifier] = {
     ...actionInfo,
     customShortCuts: storedActionInfo.customShortCuts,
+    sortIndex: sortIndex,
```

Comment:
I think the caller should (optionally) be able to set the sortIndex itself. I would set `actionInfo.sortIndex` only when `actionInfo.sortIndex` is undefined.

Reply:
I added it `sortIndex: actionInfo.sortIndex === undefined ? sortIndex : actionInfo.sortIndex,` and used it for "action.add-component" -> so that "Add Component" and "Decompose Component" are below each other.